circle.yml

```diff
@@ -38,6 +40,14 @@ dependencies:
     # create a version.json
     - printf '{"commit":"%s","version":"%s","source":"https://github.com/%s/%s"}\n' "$CIRCLE_SHA1" "$CIRCLE_TAG" "$CIRCLE_PROJECT_USERNAME" "$CIRCLE_PROJECT_REPONAME" > version.json
 
+    # build addon
+    - cd addon/
```

Comment:
ugh, I've been trying to debug this for a while now. I didn't come across
that advice :) Thanks!

On Mon, Mar 7, 2016 at 1:59 PM, Les Orchard <notifications@github.com>
wrote:

> In circle.yml
> <https://github.com/mozilla/testpilot/pull/451#discussion_r55280188>:
>
> > @@ -38,6 +40,14 @@ dependencies:
> >      # create a version.json
> >      - printf '{"commit":"%s","version":"%s","source":"https://github.com/%s/%s"}\n' "$CIRCLE_SHA1" "$CIRCLE_TAG" "$CIRCLE_PROJECT_USERNAME" "$CIRCLE_PROJECT_REPONAME" > version.json
> >
> > +    # build addon
> > +    - cd addon/
>
> Something else I think I just learned after some googling of other folks'
> CircleCI usage: Each of these entries runs in the root directory of the
> project. It's not like a continuous shell script - each line is its own
> separate script. So, cd is undone right after you issue it.
>
> Probably needs an ugly one-liner something more like cd addon && npm
> config set spin false && npm install && npm run lint && npm run sign
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mozilla/testpilot/pull/451/files#r55280188>.
>